web/pdf_attachment_viewer.js

```diff
@@ -51,16 +51,24 @@ var PDFAttachmentViewer = (function PDFAttachmentViewerClosure() {
     this.container = options.container;
     this.eventBus = options.eventBus;
     this.downloadManager = options.downloadManager;
+
+    this._renderedCapability = pdfjsLib.createPromiseCapability();
+    this.eventBus.on('fileattachmentannotation',
+      this._appendAttachment.bind(this));
   }
 
   PDFAttachmentViewer.prototype = {
-    reset: function PDFAttachmentViewer_reset() {
+    reset: function PDFAttachmentViewer_reset(keepRenderedCapability) {
       this.attachments = null;
 
       var container = this.container;
       while (container.firstChild) {
         container.removeChild(container.firstChild);
       }
+
+      if (!keepRenderedCapability) {
```

Comment:
*Just to make sure that we're on the same page, a bit of background first:*
I wanted to ensure that what `PDFDocumentProxy_getAttachments` returns always takes precedence over any file attachments, to not change the current behavior. This is why the `_renderedCapability` is used, to ensure that we wait for the "regular" attachments to load before attempting to add any existing file attachments.

Now, when resetting `PDFAttachmentViewer` we generally want replace the `_renderedCapability` to avoid the situation where opening a new document (in the viewer) could lead to attachments from the previous document being applied to the new one (given the async nature of Promises).
However, since the new `_appendAttachments` method calls `render` which will reset the attachment view (see [pdf_attachment_viewer.js#L96](https://github.com/mozilla/pdf.js/blob/a917443ae6733e77ec0011097ddfb23fa0e740fc/web/pdf_attachment_viewer.js#L96)) to ensure that everything is rendered correctly and in alphabetic order, replacing the `_rendererCapability` here could cause pending `_appendAttachments` to incorrectly be skipped.
Hence why we need keep the `_renderedCapability` in this case.

Since this isn't an option that anyone should pass in when using `PDFAttachmentViewer.reset();` from outside (as in e.g. https://github.com/mozilla/pdf.js/blob/a917443ae6733e77ec0011097ddfb23fa0e740fc/web/app.js#L595), I didn't want to add JSDocs for it. However, I'll try to add some sort of inline comment here, to briefly explain why it's there.

Sorry, this became a bit long. Hopefully it's enough to answer your question!